client/src/components/AuthContext.tsx

```diff
@@ -1,10 +1,33 @@
-import { createContext, FC, PropsWithChildren, useContext } from "react";
+import { createContext, FC, PropsWithChildren, useContext, useEffect, useState } from "react";
+import { useQueryClient } from "@tanstack/react-query";
 
-const AuthContext = createContext({authing: true, loggedIn: false});
+// CSW - seems like this needs a way to set logged in on login success...? Trying to add 
+// that (I admit I'm leaning on chatGPT for some guidance here; ther AI suggestion needed
+// some revision here before I got this working)
 
-export const AuthContextProvider: FC<PropsWithChildren<{loggedIn:boolean; authing: boolean}>> = ({children, authing, loggedIn}) => (
-  <AuthContext.Provider value={{loggedIn, authing}}>{children}</AuthContext.Provider>
-);
+const AuthContext = createContext({
+  authing: true,
+  loggedIn: false, 
+  setLoggedInState: (loggedInState: boolean) => {},
+  setAuthingState: (loggedInState: boolean) => {} 
+});
+
+export const AuthContextProvider: FC<PropsWithChildren<{loggedIn:boolean; authing: boolean}>> = ({children}) => {
+  const [authing, setAuthing] = useState(true);
+  const [loggedIn, setLoggedIn] = useState(false);
+
+  const setLoggedInState = (loggedInState: boolean) => {
```

Comment:
If an IC were making this PR, I would expect `setLoggedInState` and `setAuthingState` handlers to retain the same reference in memory every render.

This can be accomplished by using react's `useCallback` hook. 
Also, the `setAuthing` and `setLoggedIn` could be passed directly to the 
```<AuthContext.Provider value={{ ..., setAuthing, setLoggedIn }} />```

As an engineering manager, being able to convey to IC about memory usage and performance.